src/include/btmem.h

```diff
@@ -1082,12 +1082,16 @@ struct __wt_ref {
                                   /* AUTOMATIC FLAG VALUE GENERATION STOP 8 */
     uint8_t flags;
 
-#define WT_REF_DISK 0                 /* Page is on disk */
-#define WT_REF_DELETED 1              /* Page is on disk, but deleted */
-#define WT_REF_LOCKED 2               /* Page locked for exclusive access */
-#define WT_REF_MEM 3                  /* Page is in cache and valid */
-#define WT_REF_SPLIT 4                /* Parent page split (WT_REF dead) */
-    wt_shared volatile uint8_t state; /* Page state */
+#define WT_REF_DISK 0    /* Page is on disk */
+#define WT_REF_DELETED 1 /* Page is on disk, but deleted */
+#define WT_REF_LOCKED 2  /* Page locked for exclusive access */
+#define WT_REF_MEM 3     /* Page is in cache and valid */
+#define WT_REF_SPLIT 4   /* Parent page split (WT_REF dead) */
+    /*
+     * Page state: Obscure the field name as this field shouldn't be accessed directly. This public
```

Comment:
This comment is good but needs some fixing up:
```
  /*
     * Ref state: Obscure the field name as this field shouldn't be accessed directly. The public
     * interface is made up of five functions:
     *  - __wt_ref_get_state 
     *  - WT_REF_SET_STATE
     *  - WT_REF_CAS_STATE
     *  - __wt_ref_lock
     *  - WT_REF_UNLOCK
     *
     * For more details on these functions see ref_inline.h.
     */
```

Reply:
Updated in [1717dbe](https://github.com/wiredtiger/wiredtiger/pull/10345/commits/1717dbe030dab0ddbb2e7ab981c4e95e6a65cfd0).
I don't want to do so in this PR, but should `__wt_ref_lock` also become a macro? It calls `WT_REF_CAS_STATE` which uses the `__LINE__` and `__FUNCTION__` macros, so we're currently not reporting the ref_lock call site in our ref tracking

Reply:
I think its fine as is, at the very least a user of tracking will see that the ref went through one of the places that called `__wt_ref_lock`. Personally I don't think ref tracking get used that much so we shouldn't over index on it.